src/main/java/fr/sncf/osrd/infra/railscript/value/RSType.java

```diff
@@ -8,11 +9,46 @@
     ASPECT_SET(null),
     SIGNAL(null),
     ROUTE(RouteStatus.class),
+    OPTIONAL_SIGNAL(null),
+    OPTIONAL_ROUTE(null),
     SWITCH(SwitchPosition.class);
 
     public final Class<? extends Enum<?>> enumClass;
 
     RSType(Class<? extends Enum<?>> enumClass) {
         this.enumClass = enumClass;
     }
+
+    /** Converts the value to the matching string */
+    public String toString() {
+        switch (this) {
+            case BOOLEAN:
+                return "bool";
+            case ASPECT_SET:
+                return "AspectSet";
+            case SIGNAL:
+                return "Signal";
+            case ROUTE:
+                return "Route";
+            case SWITCH:
+                return "Switch";
+            case OPTIONAL_ROUTE:
+                return "Optional route";
+            case OPTIONAL_SIGNAL:
+                return "Optional signal";
+            default:
+                return "";
+        }
+    }
+
+    /** Returns the subtype if this is an optional */
+    public RSType subType() throws InvalidInfraException {
```

Comment:
subType is sort of confusing, it's more like a contained type: ROUTE doesn't inherit from OPTIONAL_ROUTE